### ISSUE TITLE
`code-block-tools`: add `on-missing-language-tag` for unlabeled fenced code blocks

### ISSUE THREAD

**Issue by pygarap** [rvben/rumdl#889]
## Summary

Add an `on-missing-language-tag` option to Code Block Tools to control what happens when a fenced code block has no language tag.

Code Block Tools currently skips these blocks silently before language configuration is evaluated. This leaves a coverage gap even when `on-missing-language-definition` is configured strictly.

## Proposed behavior

```toml
[code-block-tools]
on-missing-language-tag = "warn"
```

| Value | Behavior |
| --- | --- |
| `"warn"` | Skip unlabeled fenced code blocks and emit one config warning per run (default) |
| `"ignore"` | Silently skip unlabeled fenced code blocks |
| `"fail"` | Record an error for each affected block, continue processing, and exit non-zero at the end |
| `"fail-fast"` | Stop immediately on the first affected block and exit non-zero |

This should apply whenever Code Block Tools processes fenced blocks through `rumdl check`, `rumdl check --fix`, or `rumdl fmt`, including `--only-code-block-tools`.

This is distinct from `on-missing-language-definition`, which handles blocks that have a language tag but no configured tool definition. It is also useful independently of MD040 because Code Block Tools can run while regular Markdown rules are disabled.

## Usage

For strict Code Block Tools coverage:

```toml
[code-block-tools]
enabled = true
on-missing-language-tag = "fail"
on-missing-language-definition = "fail"
on-missing-tool-binary = "fail-fast"

[code-block-tools.languages]
python = { lint = ["ruff:check"], format = ["ruff:format"] }
```

Given an unlabeled fenced block:

```markdown
# Example

~~~
print("Hello")
~~~
```

Run:

```console
$ rumdl check docs/
```

With `on-missing-language-tag = "fail"`, the block should produce a Code Block Tools error instead of being silently skipped.

## References

- https://github.com/rvben/rumdl/blob/main/docs/code-block-tools.md
- https://github.com/rvben/rumdl/blob/main/docs/md040.md
- #374
- #458
